### PR TITLE
Mimir implementation of DataSourceConfigs

### PR DESCRIPTION
**Pull request by wemrysi**

### CODE REVIEW

**Comment by wemrysi** [comment]
This is failing to build due to a `jawn` conflict that should be resolved once #3646 is merged.

File: mimir/src/main/scala/quasar/mimir/datasources/MimirDataSourceConfigs.scala [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      RValue.rField(NameField)
+        .set(Some(CString(name.value)))
+        .apply(dataSourceConfigP(config))
+
+    absorbError(for {
+      noName <- without(name.wrapNel)
+
+      added = noName ++ Stream(namedCfg.toJValue)
+
+      _ <- precog.ingest(tablePath, added)
```
Comment (alissapajer) [inline mimir/src/main/scala/quasar/mimir/datasources/MimirDataSourceConfigs.scala]: :tada:

File: mimir/src/main/scala/quasar/mimir/datasources/MimirDataSourceConfigs.scala [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+      slicesToStream(t.transform(Filter(TransSpec1.Id, tspec)).slices)
+        .filter(_ =/= JUndefined)
+    }
+  }
+}
+
+object MimirDataSourceConfigs {
+  val NameField = "name"
+  val TypeField = "type"
+  val TypeVersionField = "type_version"
```
[the remark this answers is not in the document]
Comment (alissapajer) [reply mimir/src/main/scala/quasar/mimir/datasources/MimirDataSourceConfigs.scala]: Oh, we are versioning the data source types. What information or metadata related to a data source might change that would warrant a bump in version?

[the remark this answers is not in the document]
Comment (wemrysi) [reply mimir/src/main/scala/quasar/mimir/datasources/MimirDataSourceConfigs.scala]: @alissapajer The configuration format for each datasource is now arbitrary JSON, with a format dictated by the individual datasources. The version is to allow for a datasource to evolve their format over time.

**Pull request merged**